### PR TITLE
add vnet peering facts module

### PR DESCRIPTION
**Pull request by yungezz**
##### SUMMARY

add azure_rm_virtualnetworkpeering_facts module

##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Review by bcoca — CHANGES_REQUESTED** [review]
this module does NOT return `ansible_facts` so it should not be a _facts module

File: lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py [hunk 1]
```diff
@@ -0,0 +1,244 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Yunge Zhu <yungez@microsoft.com>
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py]: (@yungezz)

File: lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+
+options:
+    resource_group:
+        description:
+            - Name of a resource group where the vnet exists.
+        required: True
+    name:
+        description:
+            - Name of the virtual network peering.
+    virtual_network:
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py:27]: is that actually correct?
virtual network peering is a subresource of virtual network, so I think this param should be just a **virtual_network_name**
The description looks like description for **virtual_remote_network**

File: lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+author:
+    - Yunge Zhu (@yungezz)
+'''
+
+EXAMPLES = '''
+    - name: Get virtual network peering by name
+      azure_rm_virtualnetworkpeering_facts:
+        resource_group: myResourceGroup
+        name: vnet_peer1
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py]: i would move name param to the end, just for consistency.
ALso we could call it **myVnetPeering**

File: lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+            "Get Virtual Network Peering {0}".format(self.name))
+        results = []
+        try:
+            response = self.network_client.virtual_network_peerings.get(resource_group_name=self.resource_group,
+                                                                        virtual_network_name=self.virtual_network['name'],
+                                                                        virtual_network_peering_name=self.name)
+            self.log("Response : {0}".format(response))
+            results.append(vnetpeering_to_dict(response))
+        except CloudError:
+            self.log('Did not find the Virtual Network Peering.')
```
Comment (zikalino) [inline lib/ansible/modules/cloud/azure/azure_rm_virtualnetworkpeering_facts.py:212]: fail?

**Review by zikalino — CHANGES_REQUESTED** [review]
one major thing to change is **virtual_network** parameter.
others are minor

**Comment by zikalino** [comment]
@bcoca this is something we have been discussing with @nitzmahone and @jborean93 . 
At some point we have agreed that cloud resource related modules should not stick facts into **ansible_facts**.
Apparently facts should apply to the host where playbook is executed, but in case of azure_rm_*_facts are always executed on localhost.

**Comment by nitzmahone** [comment]
Policy on `_facts` will change in 2.9 (changing to `_info` and we'll alias), but for now, it's not a blocker to merge for 2.8.

**Comment by yungezz** [comment]
ready_for_review

**Comment by yungezz** [comment]
HI  @bcoca  could you pls look at the pr  again?  it's  updated and discussed with Matt about the _facts.

**Review by zikalino — APPROVED** [review]

**Comment by zikalino** [comment]
shipit

**Comment by Fred-sun** [comment]
@bcoca Please help re-review this PR when you're available! Thanks!

**Review by bcoca — CHANGES_REQUESTED** [review]
these are still improperly named _facts while not returning ansible_facts, the rename to _info is a new requirement and not a blocker, but them being improperly named _facts is a blocker.

**Review by nitzmahone — APPROVED** [review]
now that https://github.com/ansible/ansible/issues/54280#issuecomment-479171217 has been settled, good to merge.

**Comment by yungezz** [comment]
rebuild_merge

**Comment by yungezz** [comment]
per IRC core meeting result as commented in https://github.com/ansible/ansible/issues/54280#issuecomment-479172664, keep this module as _facts for 2.8. and update it later.

**Pull request merged**